dev/ledger.md

```diff
@@ -523,6 +535,100 @@ valid vote round in its participation key is strictly less than the current roun
+
+- The block's _round_, under msgpack key `r`.
+
+- The block's SHA256 Transaction commitment, under msgpack key `tc`.
+
+# Light Block Header Commitment
+
+Light Block Header Commitment for rounds (_X_$\cdot$$\delta_{SP}$,...,(_X_+1)$\cdot$$\delta_{SP}$] for some number _X_, defined as
+the root of a vector commitment whose leaves are light block headers for rounds  _X_$\cdot$$\delta_{SP}$,...,(_X_+1)$\cdot$$\delta_{SP}$ respectively. We use SHA256 hash function to create this vector commitment.
+
+# State Proof message
+
+A state proof message for rounds (_X_$\cdot$$\delta_{SP}$,...,(_X_+1)$\cdot$$\delta_{SP}$] for some number _X_, 
+contains the following components:
+
+ - Light block headers commitment for rounds (_X_$\cdot$$\delta_{SP}$,...,(_X_+1)$\cdot$$\delta_{SP}$], under msgpack key `b`.
+
+ - First attested round which would be equal to _X_$\cdot$$\delta_{SP}$ + 1, under msgpack key `f`. 
+
+ - Last attested round which would be equal to (_X_+1)$\cdot$$\delta_{SP}$, under msgpack key `l`. 
+
+ - Participant commitment used to verify state proof for rounds ((_X_+1)$\cdot$$\delta_{SP}$,...,(_X_+2)$\cdot$$\delta_{SP}$], 
+   under msgpack key `v`. 
+
+ - The $\ln(ProvenWeight)$ with 16 bits of precision that would used to verify state proof for rounds ((_X_+1)$\cdot$$\delta_{SP}$,...,(_X_+2)$\cdot$$\delta_{SP}$], under msgpack key `P`. 
+
+# State Proof Tracking
+
+Each block header keeps track of the state needed to construct, validate,
+and record state proofs.  
+This tracking data is stored in a map under the msgpack key `spt` in the block header.
+The map is indexed by the type of the state proof; at the moment, only
+type 0 is supported.  In the future, other types of state proofs
+might be added.
+
+For type 0, $KQ_{SP}=256$, $f_{SP}$ is $2^{32}*30/100$
+(as the numerator of a fraction out of $2^{32}$), $N_{SP}=1024$,
+$\delta_{SP}=256$, $\delta_{SPR}=10$  and $\delta_{SPB}=16$ .
+
+The value of the tracking data is a msgpack map with three
+elements:
+
+- Under key `n`, the next expected round of a state proof that
+  should be formed.  When upgrading from an earlier consensus protocol
+  to a protocol that supports state proofs, the `n` field is
+  set to the lowest value such that `n` is a multiple of $\delta_{SP}$
+  and so that the `n` is at least the first round of the new protocol
+  (supporting state proofs) plus $\delta_{SPB}+\delta_{SP}$.
+  This field is set in every block.
+
+- Under key `v`, the root of the Vector commitment to an array of
+  participants that are eligible to vote in the state proof at
+  round $\delta_{SP}$ from the current block.  Only blocks whose round
+  number is a multiple of $\delta_{SP}$ have a non-zero `v` field.
+
+- Under key `t`, the total weight of participants in the Vector commitment.
+
+The participants committed to by the Vector commitment are chosen in a
+specific fashion:
+
+- First off, because it takes some time to collect all of the online
```

Comment:
Does this mean that account table used for state proof is actually *more* recent than the account table used for consensus (which looks back 320 rounds)?
Can this cause issues?

My understanding is that the 320 loopback for consensus is due to sortition (to avoid accounts being able to cheat with the sortition mechanism by choosing keys or blocks in a way that would allow them to be selected more often than they should).
Is this the reason why \delta_SP can be lower than 320?

Reply:
yes. the balances are more recent than consensus. in order to verify state proof for round N we use balances from N-256-16 rounds ago.

since state proofs do not have the sortition constrain, it is better to use up-to-date keys.  Delaying when Falcon keys take effect might allow a quantum computer to defeat the post-quantum security of SPs.

Currently the state proof interval `delta_{SP}` is 256. we might not want to make this value too large since weights might shift during this interval and proof would be weaker. we can definitely go less than that if we won't have performance issues